app/controllers/issues/evidence_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Issues::EvidenceController < AuthenticatedController
+  include ContentFromTemplate
+  include DynamicFieldNamesCacher
+  include MultipleDestroy
+  include ProjectScoped
+
+  before_action :set_issues, only: [:index, :new]
+  before_action :set_affected_nodes, only: :index
+  before_action :set_columns, only: :index
+
+  def index
+    render layout: false
+  end
+
+  def new
+    @nodes_for_add_evidence = current_project.nodes.user_nodes.order(:label)
+
+    @template_content = template_content if params[:template]
+  end
+
+  private
+
+  def set_columns
+    default_field_names = ['Affected', 'Title', 'Created', 'Updated'].freeze
+    extra_field_names = ['Created by'].freeze
+
+    dynamic_fields = dynamic_field_names(@issue.evidence)
+
+    rtp = current_project.report_template_properties
+    rtp_default_fields = rtp ? rtp.issue_fields.default.field_names : []
+
+    @default_columns = rtp_default_fields.presence || default_field_names
+    @evidence_columns = default_field_names | rtp_default_fields | dynamic_fields | extra_field_names
```

Comment:
My initial instinct was to ask for a rename here for `@all_columns` which would make more sense (we're in the context of EvidenceController after all) but then I thought: not worth it.

Now I see we're going `@all_columns` in IssuesController, so have the consistency excuse to make the request.

Reply:
I'm all for "all_columns"